OpenRA.Game/Traits/TraitsInterfaces.cs

```diff
@@ -82,13 +82,30 @@ public enum ImpactType
 
 	public class AttackInfo
 	{
-		public int Damage;
+		public Damage Damage;
 		public Actor Attacker;
-		public IWarhead Warhead;
 		public DamageState DamageState;
 		public DamageState PreviousDamageState;
 	}
 
+	public class Damage
+	{
+		public int Value;
```

Comment:
On a second thought, if you're fine remaking the object there then I think I can make both of 'em readonly.